include/caffe/util/math_functions.hpp

```diff
@@ -12,6 +12,18 @@
 
 namespace caffe {
 
+template<typename Dtype>
+Dtype caffe_ipow(Dtype base, int exp) {
```

Comment:
I'm not sure that caffe_ipow is needed for this PR. The power function is only used in very few locations, so that the performance difference probably does not matter (a pow call takes 0.1 microseconds on my machine).